# **13 September 2023**

In [ ]:
import numpy as np
import pandas as pd

In [ ]:
data= pd.read_csv('https://raw.githubusercontent.com/ardhiraka/PFDS_sources/master/nbaallelo.csv')
data.head(3)

gameorder       game_id lg_id  _iscopy  year_id  date_game  seasongame  \
0          1  194611010TRH   NBA        0     1947  11/1/1946           1   
1          1  194611010TRH   NBA        1     1947  11/1/1946           1   
2          2  194611020CHS   NBA        0     1947  11/2/1946           1   

   is_playoffs team_id  fran_id  ...  win_equiv  opp_id  opp_fran  opp_pts  \
0            0     TRH  Huskies  ...       40.3     NYK    Knicks       68   
1            0     NYK   Knicks  ...       41.7     TRH   Huskies       66   
2            0     CHS    Stags  ...       42.0     NYK    Knicks       47   

  opp_elo_i opp_elo_n  game_location  game_result  forecast notes  
0    1300.0    1306.7              H            L       0.6   NaN  
1    1300.0    1293.3              A            W       0.4   NaN  
2    1306.7    1297.1              H            W       0.6   NaN  

[3 rows x 23 columns]

**Melihat berapa banyak data menggunakan .shape atau len()**

In [ ]:
len(data)
data.shape

(126314, 23)

**Agar tabel terlihat semuanya menggunakan pd.set_option("display.precision", none)**

none berisi 6 desimal dan boleh diubah

In [ ]:
pd.set_option("display.precision", 1)  # Settingan sudah selesai


# Pastikan tabel sudah tersetting
data.head(3)

gameorder       game_id lg_id  _iscopy  year_id  date_game  seasongame  \
0          1  194611010TRH   NBA        0     1947  11/1/1946           1   
1          1  194611010TRH   NBA        1     1947  11/1/1946           1   
2          2  194611020CHS   NBA        0     1947  11/2/1946           1   

   is_playoffs team_id  fran_id  ...  win_equiv  opp_id  opp_fran  opp_pts  \
0            0     TRH  Huskies  ...       40.3     NYK    Knicks       68   
1            0     NYK   Knicks  ...       41.7     TRH   Huskies       66   
2            0     CHS    Stags  ...       42.0     NYK    Knicks       47   

  opp_elo_i opp_elo_n  game_location  game_result  forecast notes  
0    1300.0    1306.7              H            L       0.6   NaN  
1    1300.0    1293.3              A            W       0.4   NaN  
2    1306.7    1297.1              H            W       0.6   NaN  

[3 rows x 23 columns]

**Jika ingin melihat informasi dari tabel secara keseluruhan dapat menggunakan data.info()**

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 126314 entries, 0 to 126313
Data columns (total 23 columns):
 #   Column         Non-Null Count   Dtype  
---  ------         --------------   -----  
 0   gameorder      126314 non-null  int64  
 1   game_id        126314 non-null  object 
 2   lg_id          126314 non-null  object 
 3   _iscopy        126314 non-null  int64  
 4   year_id        126314 non-null  int64  
 5   date_game      126314 non-null  object 
 6   seasongame     126314 non-null  int64  
 7   is_playoffs    126314 non-null  int64  
 8   team_id        126314 non-null  object 
 9   fran_id        126314 non-null  object 
 10  pts            126314 non-null  int64  
 11  elo_i          126314 non-null  float64
 12  elo_n          126314 non-null  float64
 13  win_equiv      126314 non-null  float64
 14  opp_id         126314 non-null  object 
 15  opp_fran       126314 non-null  object 
 16  opp_pts        126314 non-null  int64  
 17  opp_elo_i      126314 non-nul

**Melihat overview dari tabel menggunakan .describe()**

In [ ]:
data.describe()

gameorder   _iscopy   year_id  seasongame  is_playoffs       pts  \
count   126314.0  126314.0  126314.0    126314.0      1.3e+05  126314.0   
mean     31579.0       0.5    1988.2        43.5      6.4e-02     102.7   
std      18231.9       0.5      17.6        25.4      2.4e-01      14.8   
min          1.0       0.0    1947.0         1.0      0.0e+00       0.0   
25%      15790.0       0.0    1975.0        22.0      0.0e+00      93.0   
50%      31579.0       0.5    1990.0        43.0      0.0e+00     103.0   
75%      47368.0       1.0    2003.0        65.0      0.0e+00     112.0   
max      63157.0       1.0    2015.0       108.0      1.0e+00     186.0   

          elo_i     elo_n  win_equiv   opp_pts  opp_elo_i  opp_elo_n  forecast  
count  126314.0  126314.0   126314.0  126314.0   126314.0   126314.0   1.3e+05  
mean     1495.2    1495.2       41.7     102.7     1495.2     1495.2   5.0e-01  
std       112.1     112.5       10.6      14.8      112.1      112.5   2.2e-01  
min      1091.6    1085.8       10.2       0.0     1091.6     1085.8   2.0e-02  
25%      1417.2    1417.0       34.1      93.0     1417.2     1417.0   3.3e-01  
50%      1500.9    1501.0       42.1     103.0     1500.9     1501.0   5.0e-01  
75%      1576.1    1576.3       49.6     112.0     1576.1     1576.3   6.7e-01  
max      1853.1    1853.1       71.1     186.0     1853.1     1853.1   9.8e-01

**Exploring Datasets**

- Melihat seberapa sering suatu nilai muncul dalam data menggunakan .value_counts()  => namatabel["Label"]..value_counts()

In [ ]:
data["team_id"].value_counts()

BOS    5997
NYK    5769
LAL    5078
DET    4985
PHI    4533
       ... 
INJ      60
PIT      60
DTF      60
TRH      60
SDS      11
Name: team_id, Length: 104, dtype: int64

In [ ]:
data["fran_id"].value_counts()

Lakers          6024
Celtics         5997
Knicks          5769
Warriors        5657
Pistons         5650
Sixers          5644
Hawks           5572
Kings           5475
Wizards         4582
Spurs           4309
Bulls           4307
Pacers          4227
Thunder         4178
Rockets         4154
Nuggets         4120
Nets            4106
Suns            4080
Bucks           4034
Trailblazers    3870
Cavaliers       3810
Clippers        3733
Jazz            3555
Mavericks       3013
Heat            2371
Pelicans        2254
Magic           2207
Timberwolves    2131
Grizzlies       1657
Raptors         1634
Hornets          894
Colonels         846
Squires          799
Spirits          777
Stars            756
Sounds           697
Baltimore        467
Floridians       440
Condors          430
Capitols         291
Olympians        282
Sails            274
Stags            260
Bombers          249
Steamrollers     168
Packers           72
Redskins          65
Rebels            63
Waterloo     

Tim "Lakers" memainkan sebanyak 6024 pertandingan. Tetapi hanya 5078 yang dimainkan oleh Los Angeles Lakers(LAL). Selanjutnya mencari tahu siapa yang memanikan selisih tersebut.

In [ ]:
data.loc[data["fran_id"] == "Lakers", "team_id"].value_counts()

LAL    5078
MNL     946
Name: team_id, dtype: int64

Berdasarkan hasil di atas, diketahui bahwa
Tim lakers yang memainkan pertandingan yaitu:
  - Los Angeles Lakers(LAL) sebanyak 5078.
  - Minneapolis Lakers (MNL) sebanyak 97 pertandingan




  Lalu, jika Kita juga bisa mengetahui waktu MNL memainkan pertandingan tersebut.

In [ ]:
# Pertandingan paling awal yang dilakukan

data.loc[data["team_id"]=="MNL", "date_game"].min()

# Pertandingan paling akhir yang dilakukan

data.loc[data["team_id"]=="MNL", "date_game"].max()


# Pertandingan paling awal dan akhir

data.loc[data["team_id"]=="MNL", "date_game"].agg(("min","max"))

min    1/1/1949
max    4/9/1959
Name: date_game, dtype: object

Membedah alasan tim **BOS** (Boston Celtics) memainkan game paling banyak


- Tim **BOS** paling sering menang

In [ ]:
# Paling sering menang

data.loc[data["game_result"] == "W", "team_id"].value_counts()

BOS    3517
LAL    3134
NYK    2855
DET    2430
PHI    2306
       ... 
CHP      18
INJ      18
PIT      15
DNN      11
SDS       3
Name: team_id, Length: 104, dtype: int64

In [ ]:
# Mencetak score sebanyak
data.loc[data["team_id"] == "BOS", "pts"].sum()


626484